lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenDefinitionParticipant.java

```diff
@@ -0,0 +1,126 @@
+import org.eclipse.lsp4j.jsonrpc.CancelChecker;
+import org.eclipse.lsp4xml.dom.DOMElement;
+import org.eclipse.lsp4xml.dom.DOMNode;
+import org.eclipse.lsp4xml.services.extensions.IDefinitionParticipant;
+import org.eclipse.lsp4xml.services.extensions.IDefinitionRequest;
+
+public class MavenDefinitionParticipant implements IDefinitionParticipant {
+
+	private LocalRepositorySearcher localRepositorySearcher;
+	private MavenProjectCache cache;
+
+	public MavenDefinitionParticipant(MavenProjectCache cache, LocalRepositorySearcher localRepositorySearcher) {
+		this.cache = cache;
+		this.localRepositorySearcher = localRepositorySearcher;
+	}
+	
+	@Override
+	public void findDefinition(IDefinitionRequest request, List<LocationLink> locations, CancelChecker cancelChecker) {
+		File currentFolder = new File(URI.create(request.getXMLDocument().getTextDocument().getUri())).getParentFile();
+		DOMElement element = findInterestingElement(request.getNode());
+		if (element.getLocalName().equals("module")) {
+			File subModuleFile = new File(currentFolder, element.getFirstChild().getTextContent() + "/" + Maven.POMv4);
+			if (subModuleFile.isFile()) {
+				locations.add(toLocation(subModuleFile, element));
+			}
+			return;
+		}
+		Dependency dependency = MavenParseUtils.parseArtifact(request.getParentElement());
+		DOMNode parentNode = DOMUtils.findClosestParentNode(request, "parent");
+		if (parentNode != null && parentNode.isElement()) {
+			Optional<String> relativePath = DOMUtils.findChildElementText(element, "relativePath");
+			if (relativePath.isPresent()) {
+				File relativeFile = new File(currentFolder, relativePath.get());
+				if (relativeFile.isDirectory()) {
+					relativeFile = new File(relativeFile, Maven.POMv4);
+				}
+				if (relativeFile.isFile()) {
+					locations.add(toLocation(relativeFile, (DOMElement)parentNode));
+				}
+				return;
+			} else {
+				File relativeFile = new File(currentFolder.getParentFile(), Maven.POMv4);
+				if (match(relativeFile, dependency)) {
+					locations.add(toLocation(relativeFile, (DOMElement)parentNode));
+				}
+				return;
+			}
+		}
+		if (dependency != null) {
+			File artifactLocation = getArtifactLocation(dependency);
+			LocationLink location = toLocation(artifactLocation, element);
+			if (location != null) {
+				locations.add(location);
+			}
+			return;
+		}
+	}
+
+	private boolean match(File relativeFile, Dependency dependency) {
+		MavenProject p = cache.getSnapshotProject(relativeFile).get();
+		return p != null &&
+				p.getGroupId().equals(dependency.getGroupId()) &&
+				p.getArtifactId().equals(dependency.getArtifactId()) &&
+				p.getVersion().equals(dependency.getVersion());
+	}
+
+	private File getArtifactLocation(Dependency dependency) {
+		File localArtifact = localRepositorySearcher.findLocalFile(dependency);
+		if (localArtifact != null) {
+			return localArtifact;
+		}
+		return null;
+	}
+
+	private LocationLink toLocation(File target, DOMElement element) {
```

Comment:
Does this function need the `DOMElement element` parameter? It doesn't seem to use it.

Reply:
It should be used as 4th parameter of the LocationLink, as `originSelectionRange`.
I can fix it now or in a later commit, as you prefer.

Reply:
Probably best to do it now as I'm still reviewing (if you have a moment) :)

Reply:
Done.